src/engine/Service.h

```diff
@@ -109,11 +95,20 @@ class Service : public Operation {
   static std::optional<std::string> idToValueForValuesClause(
       const Index& index, Id id, const LocalVocab& localVocab);
 
+  // Given two child-operations of a Join-, OptionalJoin- or Minus operation,
+  // this method tries to precompute the result of one if the other one
+  // (it's sibling) is a Service operation.
+  // If rightOnly is true (used by OptionalJoin and Minus),
```

Comment:
```suggestion
  // If `rightOnly` is true (used by `OptionalJoin` and `Minus`),
```
(try to use this convention also consistently for your other comments)